### PR TITLE
Frontend Frame

### PR DESCRIPTION
**Pull request by stevenelleman**
## Terms
- `topbar`: Shows faceted identities, search bar, and settings.
- `sidebars`: Show silo and forum information, can open/collapse.
- `workspace`: Where the magic happens, man. (the teal section)

## What 
Screen Shot 2020-06-04 at 6 24 02 PM

## How  
This is designed to allow sidebar changes. Currently it can only _collapse_ and _open_, but in the future I would like to allow _dragging_ it (to change its width) and _swapping_ it with another section and a `create` button for the resource type. 

## Why 
This took a while because the `workspace`  is formatted to dynamically respond to sidebar changes without overflowing _at all_. This is critical for a non-screwy UI. 

## Tasks
- [x] Spooky mystery error: https://github.com/stevenelleman/sg/compare/se/frontend?expand=1#diff-a56f199dc8ae4389525c16962082b31eR34
- [ ] Help me think of a better directory structure. I don't love the current one, at least not yet. 

## Notes to Reviewers 
Help me solve the mystery, gang! 

## Meta 
Suggestions about process-level (PRs, reviews, testing) decisions.

### CODE REVIEW

File: services/web-frontend/src/containers/WorkspaceRow.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    this.setState(state)
+  }
+
+  render() {
+    // Why 20px off? Only when left is triggered
+    var width = window.innerWidth - (this.state.left + this.state.right);
+    // Spooky mystery: this works but why???
+    if (this.state.left === 20) {
+      width += 20;
+    }
```
Comment (stevenelleman) [inline services/web-frontend/src/containers/WorkspaceRow.js:37]: Explanation: The width of `Workspace` is being recalculated from the current width of (1) the viewport, (2) the left sidebar, and (3) the right sidebar. _For some reason_ if the left sidebar is collapsed the `Workspace` width is short by 20px. This _does not_ occur when the right sidebar is collapsed (and no correcting offset is needed - it works as intended). What is introducing this difference?

Comment (stevenelleman) [reply services/web-frontend/src/containers/WorkspaceRow.js:37]: Also just for the record the sidebars are also 20px when collapsed.

Comment (stevenelleman) [reply services/web-frontend/src/containers/WorkspaceRow.js]: Fixed - and as you would expect, it was a styling issue.

**Comment by stevenelleman** [comment]
I added eslint and fixed 3/4 of the errors. I need to investigate the rules for the other 1/4.

File: services/web-frontend/src/containers/Panel.js [hunk 2]
```diff
@@ -2,6 +2,6 @@ import React from 'react';
 
 export class Panel extends React.Component {
   render() {
-    return <div / >;
+    return <div />;
```
Comment (Laserbear) [inline services/web-frontend/src/containers/Panel.js:5]: what does this do

**Pull request merged**